### PR TITLE
expose PEBS to RTVM

### PR DESCRIPTION
**Pull request by mgcao**
To fill the requirement from TCC/VTune, expose PEBS capability to RTVM.

v2-->v3: as Anthony/Jason/Yu comments:
 revert NMI notification by INIT signal

v1-->v2: improve as Anthony comments:
 1. add a PUM_PASSTHROUGH flag to control PMU/PEBS exposure
 2. set the flag in DM if the guest VM is RTVM.

Minggui Cao (4):
  hv: revert NMI notification by INIT signal
  hv: add a flag for PMU passthrough to guest VM
  hv: expose PEBS capability and MSR as PMU_PT flag
  dm: set PMU_PT flag for CPU core partition VM

Tracked-On: #6966
Acked-by: Anthony Xu <anthony.xu@intel.com>
Signed-off-by: Minggui Cao <minggui.cao@intel.com>

### CODE REVIEW

**Comment by acrnsi** [comment]
Can one of the admins verify this patch?

**Comment by mgcao** [comment]
@wenlingz the errors seems not related with my patches.

**Comment by wenlingz** [comment]
@ywan170 @anthonyzxu @dongyaozu please help approve this PR, thanks

**Comment by wenlingz** [comment]
@ywan170 @anthonyzxu @dongyaozu please help approve this PR, thanks

File: devicemodel/core/vmmapi.c [hunk 1]
```diff
@@ -186,6 +186,7 @@ vm_create(const char *name, uint64_t req_buf, int *vcpu_num)
 		create_vm.vm_flag |= GUEST_FLAG_LAPIC_PASSTHROUGH;
 		create_vm.vm_flag |= GUEST_FLAG_RT;
 		create_vm.vm_flag |= GUEST_FLAG_IO_COMPLETION_POLLING;
+		create_vm.vm_flag |= GUEST_FLAG_PMU_PASSTHROUGH;
```
[the remark this answers is not in the document]
Comment (mgcao) [reply devicemodel/core/vmmapi.c:189]: as Anthony's suggestion, the flag of PMU PT is better separated with other flags, so add a new one; and it is default set for RTVM now. If needed, we can consider to add a ACRN-DM parameter to control it in the future.

**Review by ywan170 — APPROVED** [review]

**Comment by wenlingz** [comment]
start to run premerge test

**Review by lifeix — APPROVED** [review]

**Pull request merged**